### ISSUE TITLE
Write error for "Tried to overwrite top level content" only for erroneous cases

### ISSUE THREAD

**Issue by agrrr3** [freeorion/freeorion#4424]
Feature request
===============

Idea
----
Currently, when you write a inline named value ref (name+define) in FOCS and use it in multiple places/different content, you get an error in the log "Tried to overwrite top level content".
This should not be necessary if the named value ref does not use CurrentContent or one of the This* aliases.

I think it should be possible to check for CurrentContent usage when registering a named value ref and only write the error if CurrentContent gets used.

Gain
----
You can use inline definition of named valuerefs in more cases without getting an error in the log file.

From a scripting perspective it would be good if the name+define could stay locally close to the defining content

Discussion
----------

["Tried to overwrite top level content"](https://www.freeorion.org/forum/viewtopic.php?p=115929#p115929).

**Comment by geoffthemedio** [comment]
I think it works as it should now. If you define a named ref in a macro and it gets used in multiple places, then the internal ref will have its top level content set more than once, which it shouldn't.

**Comment by agrrr3** [comment]
i was not aware of your work

side note, you switched to no_current_content, but the magic string still gets used in the named_values.txt parser
```
parse/NamedValueRefParser.cpp:        vref->SetTopLevelContent("THERE_IS_NO_TOP_LEVEL_CONTENT");
```

i agree that it kind of works, but this is a feature request to make it work better for scripters.

Setting the top level content does not matter if the top level content does not get used in the value ref; setting it more than once (or not at all or whatever) is not a problem in this case.

one implementation which would do what i suggest;

* add a m_current_content_invariant to all ValueRefs/Conditions/Effects
* determine that similar to the other invariants - if contained contents is not invariant the containing content is also not invariant

if `SetTopLevelContent` is called when `m_top_level_content` is already set:
* set the top level content to MULTIPLE_CONTENTS
* check the m_current_content_invariant:
* if the content is invariant, set the top level content to MULTIPLE_CONTENTS and log a debug message
* if the content is not invariant,  log the current error message

**Comment by agrrr3** [comment]
@vezzra this is a content **parsing**/c++ structure issue. not sure this is the right label (but maybe it is right)

**Comment by Vezzra** [comment]
> @Vezzra this is a content **parsing**/c++ structure issue. not sure this is the right label (but maybe it is right)

Yeah, wasn't sure about the correct labelling, so I went with content scripting. But you're right, that label refers to the actual content scripts, not the parsing of the content scripts in C++. Changed the label to internal.
